cnd/src/btsieve/bitcoin/bitcoin_cache.rs

```diff
@@ -0,0 +1,94 @@
+use crate::btsieve::{BlockByHash, LatestBlock};
+use bitcoin::{hashes::sha256d, util::hash::BitcoinHash};
+use fmt::Debug;
+use futures::Future;
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use std::{fmt, fmt::Formatter, sync::Arc};
+use tokio::sync::Mutex;
+
+pub struct BitcoinCache<T> {
```

Comment:
I have a different feeling about this, but I tend to name things more specific. I don't like it when files with the same name pop up in a search because you always have to imply context then by looking at other things. If you insist I can change it, but I feel it's not too important.

Reply:
> I have a different feeling about this, but I tend to name things more specific. I don't like it when files with the same name pop up in a search because you always have to imply context then by looking at other things. If you insist I can change it, but I feel it's not too important.

I do agree with Tobin here, it is more common in Rust to add clarity to an identifier by using it through a module like `bitcoin::Cache<BitcoindConnector>`.

The search thing is slightly annoying but at least CLion does a good job of adding the context, so personally I feel like it is not that bad.

Having more concise names makes the code overall more readable because at any point in the source-code you can choose, how much context you want to add. If the name already includes the context, you'd have to do aliasing which is strictly worse IMO.